### PR TITLE
Migrate lemming to packetio + bugfixes

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: dataplane/dplanerc/interface.go [hunk 1]
```diff
@@ -983,7 +923,8 @@ func (ni *Reconciler) setupPorts(ctx context.Context) error {
 		data := &interfaceData{}
 
 		portResp, err := ni.portClient.CreatePort(ctx, &saipb.CreatePortRequest{
-			Switch: ni.switchID,
+			Switch:     ni.switchID,
+			HwLaneList: []uint32{uint32(idx)},
```
Comment (guoshiuan) [inline dataplane/dplanerc/interface.go:927]: Does this assume only one lane for each port?

File: dataplane/internal/kernel/tap.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ func NewTap(name string) (*TapInterface, error) {
 			Name: name,
 		},
 		Mode:   netlink.TUNTAP_MODE_TAP,
+		Flags:  netlink.TUNTAP_MULTI_QUEUE_DEFAULTS,
```
Comment (guoshiuan) [inline dataplane/internal/kernel/tap.go:30]: Why is this flag needed?

File: dataplane/saiserver/hostif.go [hunk 3]
```diff
@@ -284,7 +284,7 @@ func (hostif *hostif) createRemoteHostif(ctx context.Context, req *saipb.CreateH
 
 		entry = fwdconfig.TableEntryAddRequest(hostif.dataplane.ID(), portToHostifTable).
 			AppendEntry(fwdconfig.EntryDesc(fwdconfig.ExactEntry(fwdconfig.PacketFieldBytes(fwdpb.PacketFieldNum_PACKET_FIELD_NUM_PACKET_PORT_INPUT).WithUint64(nid.GetNid()))),
-				fwdconfig.Action(fwdconfig.UpdateAction(fwdpb.UpdateType_UPDATE_TYPE_SET, fwdpb.PacketFieldNum_PACKET_FIELD_NUM_HOST_PORT_ID).WithUint64Value(req.GetObjId()))).Build()
+				fwdconfig.Action(fwdconfig.UpdateAction(fwdpb.UpdateType_UPDATE_TYPE_SET, fwdpb.PacketFieldNum_PACKET_FIELD_NUM_HOST_PORT_ID).WithUint64Value(id))).Build()
```
Comment (guoshiuan) [inline dataplane/saiserver/hostif.go:287]: I understand that we are creating a new IF with a new UUID from mgr, but what's the Obj ID in the request for?

File: dataplane/dplanerc/interface.go (hunk 1 above)
Comment (DanG100) [reply dataplane/dplanerc/interface.go:927]: yes

File: dataplane/internal/kernel/tap.go (hunk 2 above)
Comment (DanG100) [reply dataplane/internal/kernel/tap.go:30]: these the "high performance" defaults per the documentation

File: dataplane/saiserver/hostif.go (hunk 3 above)
Comment (DanG100) [reply dataplane/saiserver/hostif.go:287]: every hostif is associated with a port, we store that relationship in a table

**Review by guoshiuan — APPROVED** [review]

**Pull request merged**